applications/MultilevelMonteCarloLaplacianApplication/python_scripts/multilevel_montecarlo_laplacian_analysis.py

```diff
@@ -0,0 +1,724 @@
+from __future__ import absolute_import, division #makes KratosMultiphysics backward compatible with python 2.6 and 2.7
+
+import KratosMultiphysics as Kratos
+# KratosMultiphysics.Logger.GetDefaultOutput().SetSeverity(KratosMultiphysics.Logger.Severity.WARNING) # avoid printing of Kratos things
+Kratos.Logger.GetDefaultOutput().SetSeverity(Kratos.Logger.Severity.WARNING)
+import KratosMultiphysics.MyStochasticLaplacianApplication as Poisson
+
+from analysis_stage import AnalysisStage
+
+import numpy as np
+import time
+
+from pycompss.api.task import task
+from pycompss.api.api import compss_wait_on
+from pycompss.api.parameter import *
+
+class MultilevelMonteCarloLaplacianAnalysis(AnalysisStage):
+    """Main script for MultilevelMonte Carlo Poisson simulations using the pure_diffusion solver"""
+
+
+    def __init__(self,model,parameters,sample):
+        self.sample = sample
+        super(MultilevelMonteCarloLaplacianAnalysis,self).__init__(model,parameters)
+
+            
+    def _CreateSolver(self):
+        import pure_diffusion_solver
+        solver = pure_diffusion_solver.CreateSolver(self.model,self.project_parameters["solver_settings"])
+        self.PureDiffusionSolver = solver
+        return self.PureDiffusionSolver
+
+    
+    def _GetSimulationName(self):
+        return "Multilevel Monte Carlo Poisson Analysis"
+
+    
+    def ApplyBoundaryConditions(self):
+        super(MultilevelMonteCarloLaplacianAnalysis,self).ApplyBoundaryConditions()
+        ## define the forcing function
+        for node in self.model.GetModelPart("MLMCLaplacianModelPart").Nodes:
```

Comment:
what you are doing here manually you can also do using the assign_scalar_variable_process from the core => you can specify a function there

Reply:
In this case we think that is better to define here with respect to define in the ProjectParameters file. I'll discuss with @RiccardoRossi about it.